### PR TITLE
Feat/tec 4517 ct1 migration UI polling piggyback

### PR DESCRIPTION
**Pull request by stratease**
[TEC-4517](https://theeventscalendar.atlassian.net/browse/TEC-4517)

Forked from [earlier PR](https://github.com/the-events-calendar/the-events-calendar/pull/4011).

The Migration UI poll the backend to fetch the current migration status
HTML.
This update hooks on that report request to migrate some non migrated
Events.
This processing on the back of the XHR request to the backend will start
by migrating the not yet migrated event and will, then start migrating
Events that have been claimed by Action Scheduler actions are are
pending (not yet started).

Since this processing works on the premise of the Migration UI polling
the backend, the code does not exclude Action Scheduler and will,
instead, rely on it should the user close the tab/window/browser.

Artifacts:

tests
[screencast](https://share.cleanshot.com/vE0GFM)

### CODE REVIEW

**Comment by stratease** [comment]
There was one existing bug regarding the cancellation of AS events (might only appear depending on race conditions?), and a potential issue with the `dry_run` function added to the `State` class. It was missing some phases in it's evaluation that could cause issues.

**Comment by stratease** [comment]
Added State test for dry_run

File: readme.txt [hunk 1]
```diff
@@ -246,14 +246,16 @@ Remember to always make a backup of your database and files before updating!
...
+* Fix - Avoid resetting post data in some Dive theme and plugins. [BTRIA-1397]
 * Tweak - Update the organizer website field to type URL. [TEC-4395]
 * Tweak - Update the venue website field to type URL. [TEC-4349]
 * Tweak - Add an event property for if the event is currently happening. [TBD]
 * Tweak - Create a filterable function `tec_events_get_today_button_label()` for the text on the "Today" button on calendar views. [TEC-4458]
 * Tweak - Add a filter for the link title and aria-label so they match the button text more closely. [TEC-4458]
 * Tweak - Added filters: `tec_events_today_button_label` and `tec_events_view_{view_slug}_today_button_label` to filter the output of the `tec_events_get_today_button_label` function
                          `tec_events_today_button_title` and `tec_events_{view_slug}_view_today_button_title` to filter the link title and aria-label. [TEC-4458]
-
+* Tweak - Speed up the Custom Tables v1 migration process if a browser window is open on the migration UI. [TBD]
```
Comment (bordoni) [inline readme.txt:258]: Do we have ticket for this?

**Review by bordoni — APPROVED** [review]

File: readme.txt (hunk 1 above)
Comment (stratease) [reply readme.txt:258]: https://theeventscalendar.atlassian.net/browse/TEC-4517

**Review by Camwyn — APPROVED** [review]
Ignore the readme thrash - clearing out dupes and fixing ticket numbers caused conflicts.

I fixed 😉

**Pull request merged**